spark/spark-shims/src/main/java/org/apache/zeppelin/spark/SparkVersion.java

```diff
@@ -31,8 +31,10 @@ public class SparkVersion {
 
   public static final SparkVersion SPARK_3_5_0 = SparkVersion.fromVersionString("3.5.0");
 
+  public static final SparkVersion SPARK_4_0_0 = SparkVersion.fromVersionString("4.0.0");
+
   public static final SparkVersion MIN_SUPPORTED_VERSION =  SPARK_3_2_0;
-  public static final SparkVersion UNSUPPORTED_FUTURE_VERSION = SPARK_3_5_0;
+  public static final SparkVersion UNSUPPORTED_FUTURE_VERSION = SPARK_4_0_0;
```

Comment:
3.5 will be the last feature version in Spark 3.x serials, Apache Spark master branch goes 4.0.0-SNAPSHOT now